sequencing-server/src/routes/command-expansion.ts

```
@@ -219,7 +346,250 @@ commandExpansionRouter.post('/put-expansion-set', async (req, res, next) => {
+  const activityTypeNameToTemplate: { [name: string]: Mustache } = {}
+  for (const sequenceTemplate of sequenceTemplates) {
+    let activityTypeName = sequenceTemplate.activity_type;
+
+    // by design, duplicate entries (2 templates for 1 activity type in a given model) are impossible. There is no check for it.
+    if (allActivityTypes.includes(activityTypeName)) {
+      let definition = sequenceTemplate.template_definition;
+      activityTypeNameToTemplate[activityTypeName] = new Mustache(definition);
+    }
+  }
+
+  //  6. Build ExpandedActivity for each activity, a.k.a., run the template expansion for all activities
+  const expandedActivities: {
+    [id: number]:
+    {
+      "status": string,
+      "value": ExpandedActivity<string>
+    }
+  } = {}
+
+  for (const simulatedActivityId of Object.keys(allFilteredActivities).map(Number)) {
+    if (allFilteredActivities[simulatedActivityId] && !expandedActivities[simulatedActivityId]) {
+      const simulatedActivity = allFilteredActivities[simulatedActivityId];
+      if (simulatedActivity === undefined) continue;
+      const activityTypeName = simulatedActivity.activityTypeName;
+      const currentTemplate = activityTypeNameToTemplate[activityTypeName];
+
+      // If no template for this activity type, just continue
+      if (currentTemplate) {
+        // NOTE: if I have some gibberish as a variable that's obviously not defined, there will be no error.
+        //    i.e. "CMD {{ dsvsdfs }}" expands to "CMD ".
+        currentTemplate.setLanguage(languages[0])
+        const commandString = currentTemplate.execute(stringifyActivity(simulatedActivity))
+
+        // add to results
+        expandedActivities[simulatedActivityId] = {
+          value: {
+            ...simulatedActivity,
+            expansionResult: commandString,
+            errors: [] // TODO: pass the errors, once we have the errors, if we even can
```

Does this need to be addressed? Should we have a try/catch around `execute` a few lines up to facilitate this?

---

Error handling with Handlebars should get its own ticket or should be integrated with a later templating PR.

We have silent failures if the input is missing, but _also_ loud, uncaught failures it the template is wrong somehow or a helper is called on an incorrect type. We want to make both transparent to the user somehow.

Related but separate is the lack of clear proper error propagation from `sequencing-server` to `aerie-ui`.

---

#1649